starport/services/network/request.go

```diff
@@ -0,0 +1,157 @@
+func (n Network) SubmitRequest(launchID uint64, reviewal ...Reviewal) error {
+	n.ev.Send(events.New(events.StatusOngoing, "Submitting requests..."))
+
+	messages := make([]sdk.Msg, len(reviewal))
+	for i, reviewal := range reviewal {
+		messages[i] = launchtypes.NewMsgSettleRequest(
+			n.account.Address(networkchain.SPN),
+			launchID,
+			reviewal.RequestID,
+			reviewal.IsApproved,
+		)
+	}
+
+	res, err := n.cosmos.BroadcastTx(n.account.Name, messages...)
+	if err != nil {
+		return err
+	}
+
+	var requestRes launchtypes.MsgSettleRequestResponse
+	if err := res.Decode(&requestRes); err != nil {
+		return err
+	}
+	return nil
+}
+
+// fetchRequest fetches the chain request from SPN by launch and request id
+func (n Network) fetchRequest(ctx context.Context, launchID, requestID uint64) (launchtypes.Request, error) {
+	res, err := launchtypes.NewQueryClient(n.cosmos.Context).Request(ctx, &launchtypes.QueryGetRequestRequest{
+		LaunchID:  launchID,
+		RequestID: requestID,
+	})
+	if err != nil {
+		return launchtypes.Request{}, err
+	}
+	return res.Request, err
+}
+
+// verifyAddValidatorRequest verify the validator request parameters
+func (Network) verifyAddValidatorRequest(req *launchtypes.RequestContent_GenesisValidator) error {
+	// If this is an add validator request
+	peer := req.GenesisValidator.Peer
+	valAddress := req.GenesisValidator.Address
+	consPubKey := req.GenesisValidator.ConsPubKey
+	selfDelegation := req.GenesisValidator.SelfDelegation
+
+	// Check values inside the gentx are correct
+	info, _, err := gentx.ParseGentx(req.GenesisValidator.GenTx)
+	if err != nil {
+		return fmt.Errorf("cannot parse gentx %s", err.Error())
+	}
+
+	// Check validator address
+	if valAddress != info.DelegatorAddress {
+		return fmt.Errorf(
+			"the validator address %s doesn't match the one inside the gentx %s",
+			valAddress,
+			info.DelegatorAddress,
+		)
+	}
+
+	// Check validator address
+	if res := bytes.Compare(consPubKey, info.PubKey); res != 0 {
+		return fmt.Errorf(
+			"the consensus pub key %s doesn't match the one inside the gentx %s",
+			string(consPubKey),
+			string(info.PubKey),
+		)
+	}
+
+	// Check self delegation
+	if selfDelegation.Denom != info.SelfDelegation.Denom ||
+		!selfDelegation.IsEqual(info.SelfDelegation) {
+		return fmt.Errorf(
+			"the self delegation %s doesn't match the one inside the gentx %s",
+			selfDelegation.String(),
+			info.SelfDelegation.String(),
+		)
+	}
+
+	// Check the format of the peer
+	nodeHost := strings.Split(peer, "@")
+	if len(nodeHost) != 2 ||
+		len(nodeHost[0]) == 0 ||
+		len(nodeHost[1]) == 0 {
+		return fmt.Errorf(
+			"the peer %s doesn't match the peer format <node-id>@<host>",
+			peer,
+		)
+	}
+	return nil
+}
+
+// VerifyRequests if the requests are correct and simulate them with the current launch information
+// Correctness means checks that have to be performed off-chain
+func (n Network) VerifyRequests(ctx context.Context, launchID uint64, requests []uint64) error {
```

Comment:
```suggestion
func (n Network) VerifyRequests(ctx context.Context, launchID uint64, requests ...uint64) error {
```

Reply:
The variadic function is not necessary here, because we already have the request slice constructed, if we change to variadic we need to call the method with `...`
eg: `err := n.VerifyRequests(cmd.Context(), launchID, ids...)`